src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -115,7 +117,10 @@ private void filterRemovedNetworks(Set<String> selected, CharSequence[] all) {
                 Arrays.sort(result, new Comparator<WifiConfiguration>() {
                     @Override
                     public int compare(WifiConfiguration lhs, WifiConfiguration rhs) {
-                        return lhs.SSID.compareToIgnoreCase(rhs.SSID);
+                        // See #620: There may be null-SSIDs
+                        String l = lhs != null && lhs.SSID != null ? lhs.SSID : "";
+                        String r = rhs != null && rhs.SSID != null ? rhs.SSID : "";
+                        return l.compareToIgnoreCase(r);
```

Comment:
Just didn't want to get caught by the next thing being null which never should :) But yeah, can do that.